flink-java/src/main/java/org/apache/flink/api/java/operators/CrossOperator.java

```diff
@@ -133,10 +133,6 @@ public DefaultCross(DataSet<I1> input1, DataSet<I2> input2, CrossHint hint, Stri
 					new TupleTypeInfo<Tuple2<I1, I2>>(input1.getType(), input2.getType()),
 					hint, defaultName);
 
-			if (input1 == null || input2 == null) {
-				throw new NullPointerException();
-			}
-
```

Comment:
yes, but if you used the preconditions check you could supply a useful error message, for example stating which input was actually null.